### DISCUSSION TITLE
ChatGPT

### DISCUSSION THREAD

**Discussion by cd155** [JacquesCarette/Drasil#3305] [General]
Today, I spent some time playing around with ChatGPT, 

## Experiment
so I asked: write a program in haskell such as giving a String, and return a String that only has unique characters from it

It returns:
```haskell
import Data.List (nub)

uniqueChars :: String -> String
uniqueChars = nub
```
Explanation:

We start by importing the nub function from the Data.List module. `nub` takes a list and returns a new list with all the duplicate elements removed.

Next, we define a function called `uniqueChars` that takes a `String` as its argument.

In the body of the function, we simply apply the nub function to the input `String` and return the result.

That's it! With this program, you can call the `uniqueChars` function with any `String` as its argument and it will return a new `String` containing only the unique characters from the input.

## Some thoughts
It is just very impressed it could give a really good explanation. The overall structure of the ChatGPT seems very complex. The more I use it, the more I feel that is this something we already did in Drasil. 

Anything we can learn from ChatGPT to benefit Drasil development?

**Comment by smiths** [comment]
This is impressive @cd155.  I'm not sure how closely Drasil and ChatGPT are related though.  ChatGPT uses machine learning.  With enough data it can answer a dizzying number of questions.  In my opinion, Drasil could never hope to answer as many questions.  However, ChatGPT won't always be right in its answers.  If we do Drasil correctly, its answers should be correct.  In the unfortunate instance that one of our outputs is wrong, we would know how to fix it easily.  I don't think the path to fixing incorrect AI outputs is nearly as clear.
